internal/iapl/policy.go

```diff
@@ -127,6 +127,35 @@ func NewPolicyFromFile(filePath string) (Policy, error) {
 	return NewPolicy(policy), nil
 }
 
+// NewPolicyFromFiles reads the provided file paths, merges them, and returns a new Policy.
+func NewPolicyFromFiles(filePaths []string) (Policy, error) {
+	mergedPolicy := PolicyDocument{}
+
+	for _, filePath := range filePaths {
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+
+		var filePolicy PolicyDocument
+
+		if err := yaml.NewDecoder(file).Decode(&filePolicy); err != nil {
+			return nil, err
+		}
+
+		mergedPolicy.ActionBindings = append(mergedPolicy.ActionBindings, filePolicy.ActionBindings...)
+
+		mergedPolicy.Actions = append(mergedPolicy.Actions, filePolicy.Actions...)
+
+		mergedPolicy.ResourceTypes = append(mergedPolicy.ResourceTypes, filePolicy.ResourceTypes...)
+
+		mergedPolicy.Unions = append(mergedPolicy.Unions, filePolicy.Unions...)
```

Comment:
Something that worries me, is the policy document getting extended and this function getting forgotten about.

Possibly having a method on the policy document that accepts another document and merges them so the logic is kept within the struct methods and a little closer to home feels safer? But I'm not sure how realistic my concern is. But thought I'd mention it.